### ISSUE TITLE
SCM: Allow SCM Team members to perform actions for judges

### ISSUE THREAD

**Issue by lomky** [department-of-veterans-affairs/caseflow#12424]
Update the following Judge actions to be available to Special case movement team (ie supersuers).

### Acceptance Criteria
 - [ ] Supersuers can 
   - [x] Request Cases on behalf of the Judge (This is what is blocked by #12423, can proceed on other 3 if we want to work the tickets in parallel)
   - [x] Assign ama tasks to an attorney by acting on a JudgeAssignTask assigned to a judge
   - [ ] Assign legacy tasks to an attorney by acting on a JudgeLegacyAssignTask assigned to a judge
   - [x] Reassign ama case at judge assign to another judge by acting on a JudgeAssignTask assigned to a judge
   - [ ] Reassign legacy case at judge assign to another judge by acting on a JudgeLegacyAssignTask assigned to a judge
   - [x] Reassign ama case at judge review to another judge by acting on a JudgeDecisionReviewTask assigned to a judge
 - [x] Functionality is tied to a feature flag.

Unclear if the abilities should be _limited_ to those above, or if it would be acceptable to enable more options if that is the easier path to implement.

Part of #11801

**Comment by lomky** [comment]
_[What is this graph?](https://github.com/department-of-veterans-affairs/caseflow/issues/11533)_
```
1 | 
2 | |||||
3 | ||||||||||
5 | |||
8 | 
```

Possibly implemented throught JudgeTeam functions. The logging as actual users should already work, just the assign by will need overridden. Consider breaking it into distinct PRs for the various AC.

Estimating at a 3. Possibly a 2 as it could be pretty straightforward. Repalce judge with admin, good test coverage, just adding the on behalf of. Possibly a 5 because touching a lot of places, unfamiliar to new teammates.

**Comment by hschallhorn** [comment]
I think we should also confirm that the 'whodunnit' field in the papertrail is the actual user, not the judge

**Comment by hschallhorn** [comment]
This ticket has been re-scoped to give these actions to the SCM team _ONLY_. Also, the actual user who made these changes should be credited in the assigned_by column of tasks assigned/reassigned and the whodunnit column of the versions table.

**Comment by ajspotts** [comment]
@hschallhorn Re: Assign legacy/ama tasks to an attorney by acting on a JudgeAssignTask - I had assumed that we wanted the "Assign to attorney" action in the `CaseDetailsView` return the list of the attorneys assigned to the judge.  However @lomky pointed out that alternatively returning _all_ attorneys could also be what the ticket was asking for (see gif below).  Could you weigh in to clarify? 
 
![Mar-12-2020 12-19-43](https://user-images.githubusercontent.com/18618189/76542845-fc8b2380-645b-11ea-95b8-5ee445499b7b.gif)

**Comment by hschallhorn** [comment]
I would say whichever is the easier lift. SCM users will have the full list of attorneys by selecting "Other". We can file a followup ticket if they'd like to se the judge's attorneys.

**Comment by yoomlam** [comment]
Should this be closed?

**Comment by hschallhorn** [comment]
#13726
